### ISSUE TITLE
Add Mods block recipes automatically to their respective huts

### ISSUE THREAD

**Issue by smmmadden** [ldtteam/minecolonies-features#303]
### Prerequisites 

- [x ] I am running the latest **alpha** version of MineColonies for my Minecraft version.
- [x ] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure that my issue is not covered there.
- [x ] I made sure that **this change request is not a duplicate of any existing change request.**

### Context 

- Minecraft Version: 36.0.42
- MineColonies Version: 0.13.634-ALPHA
- Structurize Version: 0.13.138-ALPHA
- Related Mods and their Versions:

## Is your feature request related to a problem?

Yes actually - some huts require a lot of recipes like the Mechanics Hut, Blacksmith, Dyer, Fletcher and Stonemason.  There are more than a hundred that have to be added for Vanilla blocks, but there are hundreds in the OOTB Mods blocks.  Too many hours are spent just on the vanilla ones so adding mod specific recipes automatically would be a huge improvement to game play.
## Enhancement or Change description

Since many huts require teaching specific recipes for its builders, it makes sense that the vanilla recipes need to be taught.  However, with the hundreds of this mods blocks being present, having to teach every one would be too time consuming and problematic (missing some and identifying what got missed), they should be added automatically to the huts as they level up.
### Notes or related things

It would also be extremely helpful to either have commands to add recipes or json/xml/toml files that can be updated to include all recipes for a given colony that can be used in place of teaching each hut individually.  Set it up once, and they all apply to a given colony ID.

---
## Viewers

* Add a thumbs-up to the change request to vote for it. This helps the request become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by ravenbuilder934** [comment]
if they were added 100% automatically, how would you decide which ones to be added per level? The lower levels don't have room to keep them all.
So if this was implemented, it'd prolly be a command or a json file like you said

**Comment by smmmadden** [comment]
This is true, I would first try to iterate over the full list (sorted by internal ID/Name) and then add based on the preset levels of each hut.  So if hutA level one holds 12 recipes, the first 12 would be added.  Then 13-64 (or whatever range the next level is) add the next set of recipes until all can be added by Level 5.  It could also be setup in a way only xx% get added per Level until Level 5 and then the remaining % would be completed.  This would make it more consistent across all huts and be a bit simpler to implement.  Not sure of the performance implications of sorting, storing (start/end IDs), but is what I was thinking.

**Comment by DesmondInc** [comment]
alternately, instead of teaching them automatically, keep the ability to teach manually, and add an ability to research the recipes in the university, allowing them to craft the researched recipes without having to be taught

**Comment by smmmadden** [comment]
I agree, it would help expand the use of Research more doing so and still retain an option for those who want to do it manually. We are on our 8th Colony on the roof of the Nether.  Having to repeat these tasks so many times is cumbersome.
